### ISSUE TITLE
Structured array properties of DataModels should not be converted to FITS_rec 

### ISSUE THREAD

**Issue by mcara** [spacetelescope/jwst#3979]
PR https://github.com/spacetelescope/jwst/pull/1754 introduced automatic conversion of structured array properties to `FITS_rec`. In my opinion, `DataModel` attributes should have the type described in corresponding schemas which is unexpected. Therefore, I think https://github.com/spacetelescope/jwst/pull/1754 should be rolled back and handling of units in FITS tables should be handled separately.

CC: @hbushouse @perrygreenfield @nden

**Comment by hbushouse** [comment]
I agree that having things converted to FITS_rec within a datamodel is a contrived solution to problem of attaching units (and any other column attributes). If someone can find a way to handle these types of column attributes that doesn't require conversion to FITS_rec (at least not within the datamodel itself - conversion should happen only when serializing to a FITS file), I'm all for reverting #1754. But we do need the column units - so that problem needs to be figured out first.

**Comment by mcara** [comment]
I agree with you that units must work. I believe it is possible but it will not be trivial and quite time consuming to do it carefully.

**Comment by jdavies-st** [comment]
Labelling this as a bug, as the result of this behavior means that any units (i.e. TUNIT) described in a FITS binary table get dropped on the floor when read in as a datamodel.

**Comment by melanieclarke** [comment]
@emolter @braingram - do you know if this is still an issue?

**Comment by braingram** [comment]
I would say this can be closed.

https://github.com/spacetelescope/stdatamodels/issues/240 detailed a way to use TUNIT keywors with stdatamodels. To copy the [last comment of that closed issue](https://github.com/spacetelescope/stdatamodels/issues/240#issuecomment-2651989979)
> This could be implemented via one of the strategies described here. However, there is currently no need for the feature and this work can be deferred indefinitely until such a need arises
